File: Leetcode/LC75/TwoSum/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> map;
        
        for(int i=0; i<n; i++) {
            int diff = target - nums[i];
            if(map.count(diff)) {
                return {map[diff], i};
            } else {
                map.insert({nums[i], i});
            }
        }
        
        return {};
    }
};

File: Leetcode/LC75/TwoSum/TwoSum.go
func twoSum(nums []int, target int) []int {
    
    indexMap := make(map[int]int)
    for idx, val := range nums {
        if dIdx, isExist := indexMap[target - val]; isExist {
            return []int{idx, dIdx}
        }
        indexMap[val] = idx
    }
    return []int{}
}
